channel/consensus_channel/consensus_channel.go

```diff
@@ -526,43 +526,21 @@ func (p *Proposal) Type() ProposalType {
 	}
 }
 
-// SetTurnNum updates the turn number on the Add or Remove proposal.
-func (p *Proposal) SetTurnNum(turnNum uint64) {
-	switch p.Type() {
-	case AddProposal:
-		{
-			p.ToAdd.turnNum = turnNum
-		}
-	case RemoveProposal:
-		{
-			p.ToRemove.turnNum = turnNum
-		}
-	}
-
-}
-
-// TurnNum returns the turn number on the Add or Remove proposal.
-func (p *Proposal) TurnNum() uint64 {
-	if p.Type() == AddProposal {
-		return p.ToAdd.turnNum
-	} else {
-		return p.ToRemove.turnNum
-	}
-}
-
 // Equal returns true if the supplied Proposal is deeply equal to the receiver, false otherwise.
 func (p *Proposal) Equal(q *Proposal) bool {
-	return p.ToAdd.equal(q.ToAdd) && p.ToRemove.equal(q.ToRemove)
+	return p.ChannelID == q.ChannelID && p.ToAdd.equal(q.ToAdd) && p.ToRemove.equal(q.ToRemove)
 }
 
 // ChannelId returns the id of the ConsensusChannel which receive the proposal.
 func (p SignedProposal) ChannelId() types.Destination {
 	return p.Proposal.ChannelID
 }
 
-// TurnNum returns the turn number of the proposal.
-func (p SignedProposal) TurnNum() uint64 {
-	return p.Proposal.TurnNum()
+// SortInfo returns the channelId and turn number so the proposal can be easily sorted.
+func (p SignedProposal) SortInfo() (types.Destination, uint64) {
```

Comment:
Using `SortInfo` instead of `TurnNum()` and `ChannelId()` makes it easier to avoid naming conflicts with public struct fields.